File: feature-venue-detail/src/main/java/com/smedialink/feature_venue_detail/venue/di/VenueDetailComponent.java
package com.smedialink.feature_venue_detail.venue.di;

import com.smedialink.feature_venue_detail.venue.view.VenueFragment;

import dagger.BindsInstance;
import dagger.Component;
import home.oleg.placesnearme.AppApiProvider;
import home.oleg.placesnearme.api.AppApi;
import home.oleg.placesnearme.scopes.FeatureScope;

@FeatureScope
@Component(dependencies = AppApi.class, modules = VenueDetailModule.class)
public interface VenueDetailComponent {
    void inject(VenueFragment target);

    @Component.Builder
    interface Builder {
        VenueDetailComponent.Builder appComponent(AppApi appApi);

        @BindsInstance
        VenueDetailComponent.Builder bind(VenueFragment target);

        VenueDetailComponent build();
    }

    final class Injector {
        private Injector() {
        }

        public static void inject(VenueFragment fragment) {
            AppApi appApi = AppApiProvider.Initializer.getAppApi(fragment);
            DaggerVenueDetailComponent.builder()
                    .appComponent(appApi)
                    .bind(fragment)
                    .build()
                    .inject(fragment);
        }
    }
}

File: feature-venue-detail/src/main/java/com/smedialink/feature_venue_detail/di/VenueDetailComponent.kt
package com.smedialink.feature_venue_detail.di

import com.smedialink.feature_venue_detail.presentation.ui.VenueFragment

import dagger.BindsInstance
import dagger.Component
import home.oleg.placesnearme.AppApiProvider
import home.oleg.placesnearme.api.AppApi
import home.oleg.placesnearme.scopes.FeatureScope

@FeatureScope
@Component(dependencies = [AppApi::class], modules = [VenueDetailModule::class])
interface VenueDetailComponent {
    fun inject(target: VenueFragment)

    @Component.Builder
    interface Builder {
        fun appComponent(appApi: AppApi): VenueDetailComponent.Builder

        @BindsInstance
        fun bind(target: VenueFragment): VenueDetailComponent.Builder

        fun build(): VenueDetailComponent
    }

    object Injector {

        fun inject(fragment: VenueFragment) {
            val appApi = AppApiProvider.Initializer.getAppApi(fragment)
            DaggerVenueDetailComponent.builder()
                    .appComponent(appApi)
                    .bind(fragment)
                    .build()
                    .inject(fragment)
        }
    }
}
